packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -465,107 +466,117 @@ export function injectIntoGlobalHook(globalObject: any): void {
       };
     }
 
-    // Here, we just want to get a reference to scheduleRefresh.
-    const oldInject = hook.inject;
-    hook.inject = function(injected) {
-      const id = oldInject.apply(this, arguments);
-      if (
-        typeof injected.scheduleRefresh === 'function' &&
-        typeof injected.setRefreshHandler === 'function'
-      ) {
-        // This version supports React Refresh.
-        helpersByRendererID.set(id, ((injected: any): RendererHelpers));
-      }
-      return id;
-    };
+    if (hook.isDisabled) {
+      // This isn't a real property on the hook, but it can be set to opt out
+      // of DevTools integration and associated warnings and logs.
+      // https://github.com/facebook/react/issues/20117
+      console.error(
+        'The installed version of React DevTools is disabled. ' +
+          'https://reactjs.org/link/react-devtools',
+      );
```

Comment:
This looks like the right type of check, but the error message isn't the clearest. It's not that React DevTools has been disabled. (It may not even have been installed.)

Maybe we could say something more like:
> Something has shimmed the React DevTools global hook (__REACT_DEVTOOLS_GLOBAL_HOOK__). Fast Refresh is not compatible with this shim and will be disabled.

Reply:
When I use `console.error(message)` the build process failed with the following message. This module has no access to the React object, so it can't use console.error() with the automatically appended stack.

Reply:
@bvaughn but if I use `throw new Error('message')` the build will be successful.

Reply:
The rest of the warning message you're referring to should say to use `console['error'](...)` instead of `console.error(...)`.

https://github.com/facebook/react/blob/13a62feab8c39bc0292eb36d636af0bb4f3a78df/scripts/babel/transform-replace-console-calls.js#L50-L63

Reply:
@bvaughn I'll submit a pull request and I'll be honored if it gets merged.

Reply:
We'er already talking on an open PR. Let's give @bbeckercscc a chance to respond to this feedback before opening another PR.

Reply:
@bvaughn I picked up the issue and @bbeckercscc submitted a pull request without any notice.

Reply:
I already have the pull request ready tho

Reply:
I didn't submit a pull request earlier because the build was failing.

Reply:
You can check the issue ID to confirm I was the one who picked up the issue.

Reply:
Go ahead alphabet-codes and take it. Sorry, I didn't notice you had picked it. I'm pretty busy right now.

Reply:
@bbeckercscc thanks, I'm really honored

Reply:
@bvaughn can you please review my pull request #20208